test/compiler/effects.jl

```diff
@@ -320,7 +272,7 @@ end
 end |> !Core.Compiler.is_consistent
 @test Base.infer_effects() do
     Ref{Int}()[]
-end |> !Core.Compiler.is_consistent
+end |> Core.Compiler.is_consistent
```

Comment:
This result is wrong, so there must be a missing `:consistent` taint on the allocation itself.

Reply:
Don't we explicitly zero-init that particular construction (because it is an Integer subtype), or something like that?

Reply:
As far as I know such fields are initialized as zero, but wasn't that introduced by #52169?

Reply:
Ah I see. I wanted to initialize this to zero or something constant, but probably couldn't as I don't see code for doing that in the constructor

Reply:
So, it's not guaranteed to initialize to zero, but with the `freeze` attribute, it always initializes with some consistent value, right? If that's the case, then I think `Ref{Int}()[]` can be `:consistent`.

Reply:
I don't think `freeze` guarantees that this value will be the same across initializations, just that a single initialization's value will not magically "transmute" across multiple uses (which `undef` can do)

Reply:
Thanks for your explanation. So we can't mark allocations with uninitialized fields `:consistent`, but we can do for uses (`getfield`). Reverting the second commit seems like the right move.